include: ipc4: copier: clean it from unrelated structs

ujfalusi opened the pull request:
Hi,

for some reason the copier.h contains the definition of `struct ipc4_base_module_cfg_ext` which it should not have as it belongs to base-config.h 
Clean this up and remove an unused enum as well from the copier.h.

ujfalusi commented on src/include/ipc4/base-config.h:
```diff
@@ -144,4 +144,19 @@ struct ipc4_output_pin_format {
+	 * the item in the array. Applies to both input and output pins.
+	 */
+	struct ipc4_input_pin_format input_pins[1];
+	struct ipc4_output_pin_format output_pins[1];
```
This part looks just awful in my view, would it make sense to replace it with
```c
uint8_t pin_formats[];
```
The `output_pins` in itself can not be referenced at all, if nb_input_pins != 1, it is not pointing where it is expected it to be pointing.

ranj063 replied:
makes sense. Can you please fix this in this PR too @ujfalusi

ujfalusi commented:
Changes since v1:
- to keep the `__attribute__((packed, aligned(4)))` fro ipc4_base_module_cfg_ext use 'git commit -n'

ujfalusi commented:
Changes since v2:
- replace the input_pins[1] / output_pins[1] from the base_cfg_ext with pin_formats[] and extend the comment to make it clear that we have the input pins followed by output pins formats behind of pin_formats[]
- Add helper to calculate base_cfg_ext size based on number of pins
- add a composite struct to combine base_cfg and base_cfg_ext for easier parsing and casting by modules.

The smart amp test module is updated to use this header, but I don't want to have unrelated patches in this PR. I will create a new PR for that later on.

kv2019i commented on src/include/ipc4/copier.h:
```diff
@@ -70,16 +70,6 @@ struct ipc4_base_module_cfg_ext {
-	/* handled inside ConfigSet of module instance */
-	IPC4_MOD_INST_ENABLE = 0x3000
-};
-
```
@mwasko @abonislawski @aborisovich @marcinszkudlinski Any plans to use these, or do we go ahead and remove?

pblaszko replied:
Support for MOD_INST_ENABLE is required for IADK module adapter API.
Support for MOD_INST_PROPS is required for both: Module API and IADK module adapter API.

It seems in SOF this implementation is missing but it is required and used at least by Win driver.
I think it should be moved to generic module header, but not removed.

pblaszko replied:
@wszypelt do we test MOD_INST_PROPS for closed-source ACE FW? do we test it for SOF? I can see the test name is check_mod_inst_props_get() in fw_02_base_fw_api/test_00_debug_api

pblaszko replied:
how about ipc4/base-config.h, the same as you do with ipc4_base_module_cfg_ext ?

ujfalusi replied:
OK, then I might just use one patch to do these.

pblaszko replied:
@ujfalusi Thank you!

pblaszko commented on src/include/ipc4/copier.h:
```diff
@@ -70,16 +70,6 @@ struct ipc4_base_module_cfg_ext {
- */
-enum ipc4_base_module_params {
-	/* handled inside LargeConfigGet of module instance */
-	IPC4_MOD_INST_PROPS  = 0xFE,
```
Please note we have below: `struct ipc4_module_instance_props`
It is binded to IPC4_MOD_INST_PROPS definition. It should be moved together.

pblaszko reviewed (changes requested):
"include: ipc4: copier: Move and update ipc4_base_module_cfg_ext to base-config.h" - commit title too long?

ujfalusi commented:
> "include: ipc4: copier: Move and update ipc4_base_module_cfg_ext to base-config.h" - commit title too long?

True, but which part can be dropped to keep the meaning?
` include: ipc4: copier: Move ipc4_base_module_cfg_ext to base-config.h`
`include: ipc4: copier: Move and update ipc4_base_module_cfg_ext`
?

pblaszko commented:
> > "include: ipc4: copier: Move and update ipc4_base_module_cfg_ext to base-config.h" - commit title too long?
> 
> True, but which part can be dropped to keep the meaning? `include: ipc4: copier: Move ipc4_base_module_cfg_ext to base-config.h` `include: ipc4: copier: Move and update ipc4_base_module_cfg_ext` ?

ok, that is not blocker for me :)

ujfalusi commented:
Changes since v3:
- one patch
- move whole lot of other structs as they do not belong to the copier.h as suggested by @pblaszko

ranj063 commented:
@ujfalusi there's a failure in the DMIC test in the internal CI. I dont think it is related to this PR but not sure if this is a recent regression

lgirdwood commented:
@wszypelt @lrudyX not expecting a failure as this is a code move, could someone check. Thanks !

ujfalusi commented:
> @ujfalusi there's a failure in the DMIC test in the internal CI. I dont think it is related to this PR but not sure if this is a recent regression

@ranj063, none of the moved headers are in use at currently, I don't think it has anything to do with this patch.

ujfalusi commented:
Changes since v4:
- rebased on main to clear and re-run the tests. DMIC failing should not have anything to do with this PR, it is just moving unused structs...

kv2019i commented:
@wszypelt Intel CI test is failing and I can't decipher the cause, all tests in logs have passed. This PR only moved header definitions around, so no failures are really expected (and builds seem fine).

wszypelt commented:
@kv2019i We have problems with one of our rvp today, I repeated the tests, now it should pass

ranj063 commented:
> @kv2019i @lgirdwood We have problems with one of our rvp today, I repeated the tests, now it should pass

@wszypelt it is still failing

wszypelt commented:
@ranj063 @ujfalusi @kv2019i @lgirdwood We still have a problem with the rvp(dmic), I'm trying to fix it as soon as I can

wszypelt commented:
@ranj063 @ujfalusi @kv2019i @lgirdwood Unfortunately, the equipment failed, we managed to fix it, I hope that now everything will work, within 30 minutes the tests should be done

wszypelt commented:
> @ranj063 @ujfalusi @kv2019i @lgirdwood we managed to fix it, I hope that now everything will work, within 30 minutes the tests should be done

all green
